### PR TITLE
TS | React

### PR DESCRIPTION
**Pull request by bilalk1**

### CODE REVIEW

File: src/selectors/user-list/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { IUserListState, IUser, IState } from '../../type';
```
Comment (InamTaj) [inline src/selectors/user-list/index.ts]: import from interfaces

File: src/saga/user/index.ts [hunk 2]
```diff
@@ -2,8 +2,9 @@ import { call, put } from 'redux-saga/effects';
 import { fetchUsersFulfill, fetchUsersError } from '../../actions/users';
 import { fetchUsers } from '../../api/user';
 import { handleError } from '../../utils';
+import { IUser } from '../../type';
```
Comment (InamTaj) [inline src/saga/user/index.ts]: import from interfaces

File: src/reducers/user/index.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+const INITIAL_STATE : IUserListState = {
+  fetching: true,
+  error: '',
+  search: '',
+  users: [],
+  sort: false,
+};
+
+export const reducer = (state : IUserListState = INITIAL_STATE , action : IUserAction) : IUserListState => {
+  // action = camel(action);
```
[the remark this answers is not in the document]
Comment (bilalk1) [reply src/reducers/user/index.ts]: Yes, it is in middleware just forgot to remove this commented code. 
One thing I need to ask to correct me if I am wrong what if we do convert case at Api level no?
i.e : 
```
 GET change to camalCase 
 POST change to snake_case
```

File: src/actions/users/index.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    IUserFetchFulfilledAction,
+    IUserFetchRejectedAction,
+    IInputSearchAction,
+    IUserFilterAction,
+    IUserFetchAction,
+    IUserSortAction,
+    IUser,
+} from '../../interfaces';
+import {
+    USER_FETCH_FULFILLED, USER_FETCH_REJECTED, INPUT_SEARCH, USER_FILTER, USER_FETCH, USER_SORT,
```
Comment (InamTaj) [inline src/actions/users/index.ts]: please line them up as one per line convention followed above.

Comment (InamTaj) [reply src/actions/users/index.ts]: In fact, just use `import * as types from 'type';` and use `types.USER_SOME_ACTION` in the usage places

File: src/components/user.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    {users.map((user : IUser) : JSX.Element => (
+         <div key={user.id}>
+         <h2>{user.login}</h2>
+        <img src={user.avatarUrl} alt='Some Faces are missing' height={100}></img>
+         <p>{user.organizationsUrl}</p>
+        <p>{user.receivedEventsUrl}</p>
+        <p>{user.reposUrl}</p>
+        <p>{user.starredUrl}</p>
+        <p>{user.subscriptionsUrl}</p>
+         <hr />
```
Comment (InamTaj) [inline src/components/user.tsx]: indentation of above lines isn't consistent, please recheck

File: src/components/user.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { IUser, IUserProps } from '../interfaces';
+
+export const User = ({ users } : IUserProps) : JSX.Element => (
```
Comment (InamTaj) [inline src/components/user.tsx]: return type should be a ReactNode

File: src/interfaces/index.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import { USER_SORT, USER_FILTER, USER_FETCH, INPUT_SEARCH, USER_FETCH_REJECTED, USER_FETCH_FULFILLED } from '../type';
```
Comment (InamTaj) [inline src/interfaces/index.ts]: import * as types

File: src/interfaces/index.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+import { ChangeEvent } from 'react';
+export interface IAction {
+    type? : string;
+    payload? : string & Array<IUser> & IError;
```
Comment (InamTaj) [inline src/interfaces/index.ts]: To make this consistent across all actions, the payload should always be an object. Not string or number.

File: src/interfaces/index.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export interface IUserAction {
+    type : typeof USER_SORT | typeof USER_FILTER | typeof USER_FETCH | typeof INPUT_SEARCH |
+    typeof USER_FETCH_FULFILLED | typeof USER_FETCH_REJECTED;
+    payload : string & Array<IUser> & IError;
```
Comment (InamTaj) [inline src/interfaces/index.ts]: should be object, instead of string

File: src/interfaces/index.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+}
+export interface IUserFilterAction {
+    type : typeof USER_FILTER;
+}
+export interface IUserFetchAction {
+    type : typeof USER_FETCH;
+}
+export interface IInputSearchAction {
+    type : typeof INPUT_SEARCH;
+    payload : string;
```
Comment (InamTaj) [inline src/interfaces/index.ts]: should be object, instead of string

File: src/interfaces/index.ts [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+}
+export interface IUserFetchFulfilledAction {
+    type : typeof USER_FETCH_FULFILLED;
+    payload : Array<IUser>;
+}
+export interface IUserFetchRejectedAction {
+    type : typeof USER_FETCH_REJECTED;
+    payload : string;
```
Comment (InamTaj) [inline src/interfaces/index.ts]: should be object, instead of string

File: src/interfaces/index.ts [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    reposUrl : string;
+    htmlUrl : string;
+    nodeId : string;
+    login : string;
+    type : string;
+    url : string;
+    id : number;
+}
+export interface IError {
+    message : string;
```
Comment (InamTaj) [inline src/interfaces/index.ts]: should be object, instead of string

File: tslint.json [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+    "no-import-side-effect": [
+      true,
+      {
+        "ignore-module": "(^rxjs/|\\.html|\\.css|\\.scss)$"
+      }
+    ],
+    "prefer-conditional-expression": false,
+    "jsx-no-multiline-js": false,
+    "jsx-no-lambda": false
+  }
```
Comment (InamTaj) [inline tslint.json]: fix EOF please

**Pull request merged**